geowrangler/raster_zonal_stats.py

```diff
@@ -98,3 +102,171 @@ def create_raster_zonal_stats(
     aoi = aoi.merge(results, how="left", left_index=True, right_index=True)
 
     return aoi
+
+# %% ../notebooks/03_raster_zonal_stats.ipynb 30
+def _validate_aggs(aggregation, band_count):
+    "Validate aggregations based on band count, dropping invalid entries"
+    aggregation_validated = []
+
+    # Iterate over the list of dictionaries
+    for agg in aggregation:
+        # Check if the 'band' value is less than or equal to band_count
+        if agg["band"] > band_count:
+            warnings.warn(
+                f"Band number in {json.dumps(agg)}  is invalid. Aggregation will be skipped."
+            )
+            continue
+        aggregation_validated.append(agg)
+
+    return aggregation_validated
+
+# %% ../notebooks/03_raster_zonal_stats.ipynb 31
+def create_exactextract_zonal_stats(
+    aoi: Union[
+        str, gpd.GeoDataFrame
+    ],  # The area of interest geodataframe, or path to the vector file
+    data: Union[str, Path],  # The path to the raster data file
+    aggregation: List[Dict[str, Any]],  # List of agg specs,
+    include_cols: List[
+        str
+    ] = None,  # If not None, list of columns from input AOI to include in output
+    include_geom: bool = True,  # If false, drop geometry column. include_cols takes priority.
+    extra_args: dict = dict(
+        strategy="feature-sequential", max_cells_in_memory=30000000
+    ),  # Extra arguments to pass to `exactextract.exact_extract(). Ignores output, include_geom, and include_cols.
```

Comment:
Just noticed this now but we may need to avoid initialized functions with `list` and `dict` variables as unintended bugs may occur (see [here](https://github.com/thinkingmachines/geo-retail-data-mart/issues/111) for more info). Will file a separate Issue for this.

Reply:
lets leave this unchanged for now